### PR TITLE
Certora CVL2 port

### PR DESCRIPTION
**Pull request by hexonaut**
Continued from https://github.com/makerdao/dss-direct-deposit/pull/110

Some spec and cli arg adjustments were required. `exec_normal_revert` seems to be timing out, but that is happening in master branch too.

### CODE REVIEW

**Comment by sunbreak1211** [comment]
There are a bunch of `assert` and `require` that haven't been migrated to the new form without parenthesis.

File: certora/D3MHub.spec [hunk 1]
```diff
@@ -1,76 +1,76 @@
...
+    function end.vow() external returns (address) envfree;
+    function plan.dai() external returns (address) envfree;
+    function pool.hub() external returns (address) envfree;
+    function pool.vat() external returns (address) envfree;
+    function pool.dai() external returns (address) envfree;
+    function pool.share() external returns (address) envfree;
+    function share.balanceOf(address) external returns (uint256) envfree;
+    function share.totalSupply() external returns (uint256) envfree;
+    function share.wards(address) external returns (uint256) envfree;
+    function _.debt() external => DISPATCHER(true);
```
Comment (sunbreak1211) [inline certora/D3MHub.spec:55]: I'm doing some tests, but it seems a bunch of these ones can be removed now that is possible to link structures.

**Comment by sunbreak1211** [comment]
> There are a bunch of `assert` and `require` that haven't been migrated to the new form without parenthesis.

@hexonaut I'm sending a PR for this soon.

File: certora/D3MHub.spec (hunk 1 above)
Comment (hexonaut) [reply certora/D3MHub.spec:55]: feel free to submit commits

Comment (hexonaut) [reply certora/D3MHub.spec:55]: feel free to submit commits

**Comment by hexonaut** [comment]
So besides the test that is timing out (can fix async) does this look good to merge now?

**Comment by sunbreak1211** [comment]
I've created this PR to clean up some unnecessary constraints: https://github.com/makerdao/dss-direct-deposit/pull/114
All the specs are passing except for the same one that is failing everywhere. However the failure has been a bit different. I just re-ran it to see if we get now the same timeout result.

Btw, are you guys planning to fix the unit tests?

**Comment by hexonaut** [comment]
Changes look good on that PR. Unit tests are fixed now and I merged in changes as you can see in CI on this PR.

Let's merge in your PR to this one, wait for CI to all pass and merge into master if there's nothing else.

**Comment by sunbreak1211** [comment]
Ohh ok great, I hadn't noticed the merge from `master` was to solve the unit tests. Let's wait until the CI finishes in the other PR and then we can merge.

**Comment by sunbreak1211** [comment]
There is something weird, why did it disappear the Makefile change?

**Comment by hexonaut** [comment]
Yeah what lol. I'll re-add.

**Comment by hexonaut** [comment]
Ah I accidentally merged master incorrectly. Anyways, fixed.

**Comment by hexonaut** [comment]
Ok looks good other than the timeout. LMK if I'm good to merge.

**Review by sunbreak1211 — APPROVED** [review]
Looks good.

**Pull request merged**